### DISCUSSION TITLE
What is the value stored in spike_count, in the NWB units field

### DISCUSSION THREAD

**Discussion by GaelleChapuis** [dandi/helpdesk#101] [Q&A]
I am trying to work with this dataset
```
dandiset_id = '000409'
filepath = 'sub-KS042/sub-KS042_ses-07dc4b76-5b93-4a03-82a0-b3d9cc73f412_behavior+ecephys+image.nwb'
```
and loaded the nwb units as such:
```
nwb = io.read()
units = nwb.units.to_dataframe()
```

I wanted to rapidly know the number of spikes for each unit, and was hoping to find this information already computed. I thought the column `spike_count` was this value, however it does not match.
For example: 
```
index_test = 2
units.spike_times[index_test].__len__()
Out: 224426
units.spike_count[index_test]
Out: 32988.0
```

Can someone explain what this column value is ?

**Comment by CodyCBakerPhD** [comment]
You can find the total number of spikes for each unit a couple of different ways, but I believe the recommended would be

```python
len(nwbfile.units.get_unit_spike_times(unit_id))
```

I will also mention for general education; the `units` table only has a few pre-defined columns, the most important being of course 'spike_times'. The other column you are observing, 'spike_count' must have been added as a custom column. For any column on any table in NWB, you can always attempt to find more information by viewing the `description`

```python
nwbfile.units["spike_count"].description
```

Now, specific to this problem in this file - after digging into this a bit it appears as if there is an indexing issue between the unit IDs as extracted from the IBL SpikeSortingLoader and the cluster metrics assigned to those IDs [[source code](https://github.com/catalystneuro/IBL-to-nwb/blob/main/ibl_to_nwb/updated_conversion/datainterfaces/iblsortingextractor.py)]

This leads to inconsistencies between the `spike_times` for each unit and the quality metrics assigned to that row on the table

I will try to isolate this index mismatch, push a fix, and reconvert+upload some files that only contain the processed (non-raw) data to resolve this temporarily while I work on the latest passive data ingestion
